GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -358,6 +364,33 @@ extension SecureConversations {
                 return
             }
         }
+
+        @discardableResult
+        private func validateMessage() -> Bool {
+            let canSendText = !messageText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
+            // TODO: File uploading validation MOB-1742
+            action?(.sendButtonHidden(!canSendText))
+            return canSendText
+        }
+
+        private func sendMessage() {
+            guard validateMessage() else { return }
+
+           _ = environment.sendSecureMessage(
+                messageText,
+                nil, // TODO: File uploading MOB-1742
+                environment.queueIds
+            ) { [weak self] result in
+                switch result {
+                case .success:
+                    self?.messageText = ""
+                    self?.delegate?(.chatEngagementRequested)
```

Comment:
Why do we request a chat engagement after sending a message?

Reply:
To allow upgrade to live chat engagement. But maybe I do not understand it correctly, please advise: what should happen when message is successfully sent?

Reply:
Removed this request. Engagement flow will be addressed separately. b6165071268d0f739f15105ba61b2521f786dbf8